### PR TITLE
GET /api/v1/appointments/:appointmentDate

### PR DESCRIPTION
**Pull request by hassanelnajjar**
# GET /api/v1/appointments/:appointmentDate

- [x] Add getAppointmentsByDate Controllers and it's query
- [x] add database query test and route test 
- [x] add date schema validation function inside utils
- [x] add boomify function to shape the error
- [x] Configure Circle CI Database
- [x] Edit FakeData and change the date type to timestampTZ

relates #16

### CODE REVIEW

File: server/database/config/index.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ const dbBuild = require('./build');
   try {
     await dbBuild();
     console.log('Build Successfully');
+    const { rows } = await connection.query('SELECT * from appointments');
+    console.log(rows);
```
Comment (Mu7ammadAbed) [inline server/database/config/index.js:10]: remove these

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: server/database/config/index.js (hunk 1 above)
Comment (hassanelnajjar) [reply server/database/config/index.js:10]: ok

File: server/controllers/appointments/getAppointmentsByDate/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    const { appointmentDate } = req.params;
+
+    const isValid = await appointmentDateValidation.isValid({
+      appointmentDate,
+    });
+    if (!isValid) {
+      throw boomify(400, 'Invalid Date', 'Please send a correct date');
+    }
+    const { rows } = await getAppointmentsByDate(appointmentDate);
+    res.json({ status: 200, message: 'success', data: rows });
```
Comment (Mu7ammadAbed) [inline server/controllers/appointments/getAppointmentsByDate/index.js]:
```suggestion
if (!isValid) {
      next(boomify(400, 'Invalid Date', 'Please send a correct date'));
    } else {
      const { rows } = await getAppointmentsByDate(appointmentDate);
      res.json({ status: 200, message: 'success', data: rows });
    }
```

Comment (hassanelnajjar) [reply server/controllers/appointments/getAppointmentsByDate/index.js]: ok

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by AlaaShurrab — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Pull request merged**